composer.json

```diff
@@ -37,7 +37,15 @@
             "library/sanitize.inc.php",
             "library/date_functions.php",
             "library/validation/validate_core.php"
-        ]
+        ],
+        "psr-4": {
+            "common\\": "common",
+            "common\\database\\": "common/database",
+            "common\\logging\\": "common/logging",
```

Comment:
Did some reading on this and I think you are right to create a separate entry for each namespace.